src/ucs/datastruct/array.c

```diff
@@ -31,6 +32,14 @@ ucs_status_t ucs_array_grow(void **buffer_p, size_t *capacity_p,
     current_capacity = *capacity_p;
     new_capacity     = ucs_max(current_capacity * UCS_ARRAY_GROW_FACTOR,
                                min_capacity);
+    new_capacity     = ucs_min(new_capacity, max_capacity);
+
+    if (min_capacity > new_capacity) {
```

Comment:
This condition would be always FALSE, because of the line above:
```
new_capacity     = ucs_min(new_capacity, max_capacity);
```
Here the `new_capacity` is what we can allocate, it never exceeds the maximum allowed.
`min_capacity` is the minimum capacity requested by the caller.
So here we check if client request cannot be fulfilled, therefore this condition.

Reply:
i suggest to remove the line above, why need to calc min and then check condition, can't we just check if requested allocation is within the limit?

Reply:
Mmm, no, I believe the line with `min` is needed.

Look, imagine you have an array of 100 elements with capacity=100 and max_capacity=127 and you want to add a new single element to it.
Then on growing, the algorithm would try to double the array size:
```
new_capacity = ucs_max(current_capacity * UCS_ARRAY_GROW_FACTOR, min_capacity); // = 200
```
200 is above the max_capacity, but there is no error here, because we can still allocate the maximum allowed (so 127). So `min` here guarantees that we don't allocate beyond max_capacity.

Reply:
so maybe first check if min_capacity>max_capacity and if yes print error?

Reply:
yes, it would also work, will change